client/jetpack-cloud/sections/agency-dashboard/sites-overview/hooks/use-install-boost.tsx

```diff
@@ -0,0 +1,111 @@
+import { useQueryClient } from '@tanstack/react-query';
+import { useTranslate } from 'i18n-calypso';
+import { useEffect, useState, useCallback } from 'react';
+import { useDispatch } from 'calypso/state';
+import useInstallPluginMutation from 'calypso/state/jetpack-agency-dashboard/hooks/use-install-plugin-mutation';
+import useRequestBoostScoreMutation from 'calypso/state/jetpack-agency-dashboard/hooks/use-request-boost-score-mutation';
+import { successNotice, errorNotice } from 'calypso/state/notices/actions';
+import type { Site } from '../types';
+
+export default function useInstallBoost(
+	siteId: number,
+	siteUrl: string,
+	queryKey: any[]
+): {
+	installBoost: () => void;
+	status: string;
+	requestBoostScore: () => void;
+} {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const queryClient = useQueryClient();
+
+	const [ status, setStatus ] = useState( 'idle' );
+
+	const { mutate: requestBoostScore, status: requestBoostStatus } = useRequestBoostScoreMutation( {
+		retry: false,
+	} );
+
+	const errorMessage = translate(
+		'Something went wrong while installing Boost. Please try again.'
+	);
+
+	const handleRequestBoostScore = useCallback( () => {
+		requestBoostScore( { site_ids: [ siteId ] } );
+	}, [ requestBoostScore, siteId ] );
+
+	const handleUpdateSites = useCallback( async () => {
+		// Cancel any current refetches, so they don't overwrite our update
+		await queryClient.cancelQueries( queryKey );
+
+		// Update to the new value
+		const updatedData = queryClient.setQueryData( queryKey, ( oldSites: any ) => {
+			return {
+				...oldSites,
+				sites: oldSites?.sites.map( ( site: Site ) => {
+					if ( site.blog_id === siteId ) {
+						return {
+							...site,
+							has_pending_boost_one_time_score: true,
+						};
+					}
+					return site;
+				} ),
+			};
+		} );
+		queryClient.setQueryData( queryKey, updatedData );
```

Comment:
Are we making two requests here? I see `setQueryData` called on line 42, and again on line 56. I'm not familiar with `react-query` though, so if I'm misreading, let me know 🙂

Reply:
Good catch 👁️